### DISCUSSION TITLE
live webcams

### DISCUSSION THREAD

**Discussion by krolaper** [ultralytics/yolov5#10075] [Q&A]
Doesn’t open any feed on the camera stream.
Can anyone have encountered and know the solution?
```
File "/home/odinson/PycharmProjects/Panning/Video/all/venv/lib/python3.10/site-packages/pafy/backend_shared.py", line 51, in extract_video_id
    raise ValueError(err % url)
ValueError: Need 11 character video id or the URL of the video. Got https://www.youtube.com/embed/X3Iwe_d5AVk
```

**Comment by krolaper** [comment]
Incorrect detection of videoid.
this is visible in the file "backend_shared.py" string
`vidid = parsedurl.path.split('/')[-1] if parsedurl.path else ''`
result: `ParseResult(scheme='https', netloc='www.youtube.com', path='/embed/X3Iwe_d5AVk', params='', query='', fragment='')`

**Comment by krolaper** [comment]
If you do not address youtube, the program code is always unable to determine the correctness of the link. Please provide an example of a working reference for the test and understanding if the code is considered correct. May be [https://stackoverflow.com/questions/4356538/how-can-i-extract-video-id-from-youtubes-link-in-python](url)

**Comment by krolaper** [comment]
Fixed it temporarily like this:
```
    if parsedurl.netloc in ('youtube.com', 'www.youtube.com', 'm.youtube.com', 'gaming.youtube.com'):
        query = parse_qs(parsedurl.query)
        if 'v' in query and idregx.match(query['v'][0]):
            return query['v'][0]
        if '/embed/' in parsedurl.path:
            return parsedurl.path.split('/')[2]
        if '/v/' in parsedurl.path:
            return parsedurl.path.split('/')[2]
```

**Reply by glenn-jocher** [reply]
@krolaper what's your inference command that was failing? I need to be able to reproduce any issues to look into this.

**Reply by krolaper** [reply]
@glenn-jocher detect.py source https://www.youtube.com/embed/X3Iwe_d5AVk
Also, please consider the topic: https://github.com/ultralytics/yolov5/discussions/10080
Because the same link is used.

**Reply by glenn-jocher** [reply]
@krolaper follow Usage instructions on README, i.e. use the same URL form:
```
python detect.py --source https://www.youtube.com/watch?v=X3Iwe_d5AVk
```

**Reply by glenn-jocher** [reply]
@krolaper updated datasets.py with correct-format usage examples in https://github.com/ultralytics/yolov5/pull/10106

**Reply by krolaper** [reply]
@glenn-jocher Poor decision. The user should not guess how the link should look, it is better to update the parser at least for youtube, not to mention popular other resources. I’ve already written above.

**Reply by glenn-jocher** [reply]
@krolaper yes it would be best if things just worked the first time. In this case this seems to be a pafy limitation rather than something caused by YOLOv5.

But we can try to patch this anyway. Can you please verify that we've got all possible YouTube URL formats in https://github.com/ultralytics/yolov5/discussions/10075#discussioncomment-4086022 and then submit a PR with this update?
